services/spam-ban/spam-banning.service.js

```diff
@@ -0,0 +1,108 @@
+const { EmbedBuilder } = require("discord.js");
+const config = require("../../config");
+
+class SpamBanningService {
+  static async handleInteraction(interaction) {
+    const message = interaction.options.getMessage("message");
+    if (message.author.bot || SpamBanningService.#isAdmin(message.member)) {
```

Comment:
Isn't the `isAdmin` check a bit redundant? We've already set the command's permission to `BanMembers`. What do you think?

Reply:
That is correct. The bot has admin right and have the capabilities of banning other team members. Or I think it does, this should prevent that

Reply:
Ah my bad, got confused there